sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -22,26 +25,32 @@
     private static final int HTTP_STATUS_TOO_MANY_REQUESTS = 429;
     private static final ChronoUnit DEFAULT_TIME_UNIT = ChronoUnit.MILLIS;
     private static final String RETRY_AFTER_MS_HEADER = "retry-after-ms";
-    private final int maxRetries;
-    private final Duration delayDuration;
+    private final RetryStrategy retryStrategy;
 
     /**
-     * Creates a RetryPolicy with the default number of retry attempts and delay between retries.
+     * Creates a {@link FixedDelay} RetryPolicy with the default number of retry attempts and delay between retries.
      */
     public RetryPolicy() {
-        this.maxRetries = DEFAULT_MAX_RETRIES;
-        this.delayDuration = Duration.of(DEFAULT_DELAY, DEFAULT_TIME_UNIT);
+        this(DEFAULT_MAX_RETRIES, Duration.of(DEFAULT_DELAY, DEFAULT_TIME_UNIT));
     }
 
     /**
-     * Creates a RetryPolicy.
+     * Creates a {@link FixedDelay} RetryPolicy.
      *
      * @param maxRetries the maximum number of retries to attempt.
-     * @param delayDuration the delay between retries
+     * @param delayDuration the delay between retries.
      */
     public RetryPolicy(int maxRetries, Duration delayDuration) {
-        this.maxRetries = maxRetries;
-        this.delayDuration = delayDuration;
+        this(new FixedDelay(maxRetries, delayDuration));
+    }
```

Comment:
I wonder if we bite the bullet and remove this API. We do not believe fixed retry policies are good, and if we removed this it would be easier to remove support for this in the future. Thoughts?

Reply:
I'm not against it, fixed delay unless prompted by the service for the delay time are very rigid.

Reply:
Looks like most of the places used default retry policy and none of them used this overload. So, I have removed it and made the default use exponential backoff as per spec.